pyiron_base/jobs/job/runfunction.py

```diff
@@ -674,14 +675,20 @@ def execute_job_with_external_executable(job):
         cores=job.server.cores, threads=job.server.threads, gpus=job.server.gpus
     )
     job_crashed, out = False, None
+
     try:
-        out = execute_subprocess(
-            executable=executable,
-            shell=shell,
-            working_directory=job.working_directory,
-            conda_environment_name=job.server.conda_environment_name,
-            conda_environment_path=job.server.conda_environment_path,
-        )
+        with TemporaryEnvironment(
+            PYIRON_SERVER_CORES=str(job.server.cores),
+            PYIRON_SERVER_THREADS=str(job.server.threads),
+            PYIRON_SERVER_GPUS=str(job.server.gpus),
+        ):
```

Comment:
This also affects the python process pyiron is running in. Maybe what we want instead is setting it directly to the subprocess https://docs.python.org/3/library/subprocess.html#popen-constructor

Reply:
Correct, but the context manager will unset/restore the variables after the block and during the execution of the script nothing will be done in the python process anyway, since we use a blocking call to Popen, so I don't expect any problems from this.

But either way will work and it really doesn't matter to me how it is implemented.

Reply:
I have a slight preference to do it in the subprocess directly to stay closer to the Python standard library: https://github.com/pyiron/pyiron_base/pull/1485